tree/ntuple/v7/inc/ROOT/RNTupleReader.hxx

```diff
@@ -324,7 +363,22 @@ public:
    }
 
    template <typename T>
-   ROOT::RNTupleDirectAccessView<T> GetDirectAccessView(std::string_view fieldName)
+   ROOT::RNTupleView<T> GetView(ROOT::DescriptorId_t fieldId, T *rawPtr, const std::type_info &ti)
+   {
+      return GetView<T>(fieldId, rawPtr, ROOT::Internal::GetRenormalizedDemangledTypeName(ti));
+   }
+
+   template <typename T>
+   ROOT::RNTupleView<T> GetView(ROOT::DescriptorId_t fieldId, T *rawPtr, std::string_view typeName)
+   {
+      static_assert(std::is_void_v<T>, "calling GetView with a type name string is only allowed for [T = void]");
+      auto field = RNTupleView<T>::CreateField(fieldId, *fSource, typeName);
+      auto range = ROOT::Internal::GetFieldRange(*field, *fSource);
+      return RNTupleView<T>(std::move(field), range, rawPtr);
+   }
+
+   template <typename T>
+   RNTupleDirectAccessView<T> GetDirectAccessView(std::string_view fieldName)
```

Comment:
I've noticed this compiler failure on Debian, probably we need `ROOT::RNtupleDirectAccessView`?